### PR TITLE
Implement server overview page

### PR DESCRIPTION
**Pull request by jgrund**
This patch implements the server overview page. It is not
feature-complete as things like adding servers, detect actions and the action-dropdown
are missing.

However, it should be good enough for a first pass. Everything should update in
near-realtime.

server-overview-changes

Signed-off-by: Joe Grund <jgrund@whamcloud.io>

---
This change is [Reviewable](https://reviewable.io/reviews/whamcloud/integrated-manager-for-lustre/1530)

### CODE REVIEW

File: iml-gui/crate/src/lib.rs [hunk 1]
```diff
@@ -47,6 +48,14 @@ const MAX_SIDE_PERCENTAGE: f32 = 35_f32;
 /// help url becomes `https://localhost:8443/help/docs/Graphical_User_Interface_9_0.html`
 const CTX_HELP: &str = "help/docs/Graphical_User_Interface_9_0.html";
 
+pub fn extract_api(s: &str) -> Option<&str> {
+    let re = Regex::new(r"^/?api/[^/]+/(\d+)/?$").unwrap();
+
+    let x = re.captures(s)?;
+
+    x.get(1).map(|x| x.as_str())
```
Comment (ip1981) [inline iml-gui/crate/src/lib.rs]: FYI, I used this for file systems:

```rust

        lazy_static! {
            static ref RE: Regex = Regex::new(r"/(\d+)/?$").unwrap();
        }
        let srv_id = RE.captures(&t.primary_server).unwrap().get(1).unwrap();
```

File: iml-gui/crate/src/lib.rs [hunk 2]
```diff
@@ -47,6 +48,14 @@ const MAX_SIDE_PERCENTAGE: f32 = 35_f32;
 /// help url becomes `https://localhost:8443/help/docs/Graphical_User_Interface_9_0.html`
 const CTX_HELP: &str = "help/docs/Graphical_User_Interface_9_0.html";
 
+pub fn extract_api(s: &str) -> Option<&str> {
```
Comment (ip1981) [inline iml-gui/crate/src/lib.rs]: Maybe name it `extract_id`?

**Comment by ip1981** [comment]
Nit: make a tooltip for things like `3 days ago` with the exact date - "2012-02-02 12:34:56 UTC+2" :)

File: iml-gui/crate/src/lib.rs (hunk 2 above)
Comment (jgrund) [reply iml-gui/crate/src/lib.rs]: I'll update in a follow-on.

**Comment by jgrund** [comment]
Merged as 4fde7b9fbb1b0112ca515b10f0a4a8016549986c

**Pull request closed without merging**